### PR TITLE
separate Secret and configmap switch case

### PR DESCRIPTION
**Pull request by Avni-Sharma**
While reading secret the data is encoded. This PR makes a separate switch case for reading the secret data.

### CODE REVIEW

File: pkg/controller/servicebindingrequest/bind_custom_resources_test.go [hunk 1]
```diff
@@ -72,3 +73,64 @@ func TestBindNonBindableResources_GetOwnedResources(t *testing.T) {
...
+func TestBindNonBindableResources_Secret_GetOwnedResources(t *testing.T) {
+	f := mocks.NewFake(t, "test")
+	cr := mocks.DatabaseCRMock("test", "test")
+	trueBool := true
+	reference := v1.OwnerReference{
+		APIVersion:         cr.APIVersion,
+		Kind:               cr.Kind,
+		Name:               cr.Name,
+		UID:                cr.UID,
+		Controller:         &trueBool,
```
Comment (isutton) [inline pkg/controller/servicebindingrequest/bind_custom_resources_test.go]: Can you create a `TruePtr` and `FalsePtr` constants (or equivalent) and make them available to be used globally?

Comment (akashshinde) [reply pkg/controller/servicebindingrequest/bind_custom_resources_test.go]: +1

File: pkg/controller/servicebindingrequest/bind_custom_resources_test.go [hunk 2]
```diff
@@ -72,3 +74,63 @@ func TestBindNonBindableResources_GetOwnedResources(t *testing.T) {
...
+				Resource: "routes",
+			},
+		},
+		f.FakeDynClient(),
+	)
+
+	t.Run("Should return secret as owned resource", func(t *testing.T) {
+		resources, err := b.GetOwnedResources()
+		require.NoError(t, err)
+		require.Equal(t, 2, len(resources), "Should return 1 owned resource")
```
Comment (akashshinde) [inline pkg/controller/servicebindingrequest/bind_custom_resources_test.go]: @Avni-Sharma should return 1 ?

File: pkg/controller/servicebindingrequest/bind_custom_resources_test.go [hunk 3]
```diff
@@ -72,3 +74,63 @@ func TestBindNonBindableResources_GetOwnedResources(t *testing.T) {
...
+	})
+
+	t.Run("Should return all variables exist in the secret data section", func(t *testing.T) {
+		data, err := b.GetBindableVariables()
+		require.NoError(t, err)
+		require.Equal(t, 3, len(data), "")
```
Comment (akashshinde) [inline pkg/controller/servicebindingrequest/bind_custom_resources_test.go]: @Avni-Sharma we should assert if the secret key and values are correct

**Pull request merged**